vision/module/module_orientation.py

```diff
@@ -0,0 +1,80 @@
+"""
+module_orientation will calculate the orientation of the module in degrees
+and return them as a tuple (x tilt, y tilt) using a derivative through the x- and y- axes
+"""
+import numpy as np
+import argparse
+from region_of_interest import region_of_interest
+
+
+def get_module_orientation(roi, coordinates):
+    """
+    Finds the orientation of the module in degrees
+
+    Parameters
+    ----------
+    roi: numpy array
+        module region of interest calculated by region_of_interest
+    coordinates: tuple of integers
+        (x, y) coordinates of the center of the module in the frame
+    """
+    x_pos, y_pos = coordinates
+
+    num_rows, num_cols = roi.shape
+    print("rows: " + str(num_rows))
+    print("cols: " + str(num_cols))
+
+    x_diffs = 0
+    for row in roi:
+        x_diffs = x_diffs + (row[num_cols-1] / 1000 - row[0] / 1000)
+        # x_diffs = x_diffs + np.mean(np.diff(row))
+
+    x_avg_diff = x_diffs / num_rows
```

Comment:
This is isometric to
```python
x_avg_diff = np.mean(roi[:, -1] - roi[:, 0]) / 1000
```
Yet this version avoids for loops, so it will be faster with numpy.

Reply:
This also applies to the y version.